src/lib/chart/dendrogram.typ

```diff
@@ -0,0 +1,312 @@
+#import "/src/draw.typ"
+#import "/src/styles.typ"
+#import "../axes.typ"
+#import "../palette.typ"
+
+#let default-style = (
+  axes: (:)
+)
+
+// Valid dendrogram modes
+#let dendrogram-modes = (
+  "vertical",
+  "horizontal"
+)
+
+// Functions for max value calculation
+#let dendrogram-max-value-fn = (
+  vertical: (data, value-key) => {
+    calc.max(0, ..data.map(t => t.at(value-key)))
+  },
+  horizontal: (data, value-key) => {
+    calc.max(0, ..data.map(t => t.at(value-key)))
+  },
+)
+
+// Functions for min value calculation
+#let dendrogram-min-value-fn = (
+  vertical: (data, value-key) => {
+    calc.min(0, ..data.map(t => t.at(value-key)))
+  },
+  horizontal: (data, value-key) => {
+    calc.min(0, ..data.map(t => t.at(value-key)))
+  },
+)
```

Comment:
From what I can tell these functions only get used once, could they be placed inline instead?

Reply:
As I don't see the min max functions changing between modes, I've changed it from a dictionary to just a function. I think it still makes sense to have them as discrete functions for code readability